framework/src/util/FrameworkFactory.cpp

```diff
@@ -40,6 +39,14 @@ struct CoreBundleContextHolder
   ~CoreBundleContextHolder()
   {
     auto const state = ctx->systemBundle->state.load();
+
+    // The framework may have already completed its stop. In this case,
+    // nothing needs to be done; the framework can be destroyed.
+    // Allowing a call to WaitForStop can cause a crash during static DLL
+    // destruction on Windows even if the framework was explicitly stopped
+    // prior to static destruction.
+    if (Bundle::STATE_INSTALLED == state || Bundle::STATE_RESOLVED == state) return;
```

Comment:
why not use unary operators and avoid the return statement in the middle of the function.
`if ((Bundle::STATE_INSTALLED | Bundle::STATE_RESOLVED) & state) `